internal/pkg/deploy/cloudformation/app.go

```diff
@@ -136,7 +137,7 @@ func (cf CloudFormation) DelegateDNSPermissions(app *config.Application, account
 		AccountID:          app.AccountID,
 		DomainName:         app.Domain,
 		DomainHostedZoneID: app.DomainHostedZoneID,
-		Version:            deploy.LatestAppTemplateVersion,
+		Version:            version.LatestTemplateVersion(),
```

Comment:
What do you think of the following structure as an alternative:
```go
// In version/version.go
var Version string
var LatestAppTemplateVersion = Version

// In internal/pkg/deploy/cloudformation/app.go
var latestAppTemplateVersion = version.LatestAppTemplateVersion
func (cf CloudFormation) DelegateDNSPermissions(app *config.Application, accountID string) error {
   deployApp := deploy.CreateAppInput{
		Name:               app.Name,
		AccountID:          app.AccountID,
		DomainName:         app.Domain,
		DomainHostedZoneID: app.DomainHostedZoneID,
		Version:            latestAppTemplateVersion,
}

// In internal/pkg/deploy/cloudformation/app.go
func TestCloudFormation_DelegateDNSPermissions(t *testing.T) {
  mockLatestAppTemplateVersion := v1.99.99
  for name, tc := range testCases {
       latestAppTemplateVersion = mockLatestAppTemplateVersion
  }
}
```

I can see how this introduces more lines of code changes, and requires individual packages to implement their own way of mocking the latest app template version - so more churning for developers.

However, the benefit is that it keeps the "mocking" logic within individual packages, instead of providing a general mock from the "source" package (in this case, the "source" package is `version`)

I am fine either way, just proposing a different idea!

Reply:
Good question. The reasons for having `LatestTemplateVersion()` are:

1. Previously it was a const so that it is immutable, and we might want to keep this benefit so that we don't accidentally modify the version number
2. To avoid the fact that each test function has to assign value to `LatestAppTemplateVersion` otherwise it'll be empty since the env var is not set for each test environment.

> However, the benefit is that it keeps the "mocking" logic within individual packages, instead of providing a general mock from the "source" package (in this case, the "source" package is version)

If a package needs its own mock, it can override this behavior I think. For example in CLI package instead of using `version.LatestTemplateVersion()`, it uses a cache variable in the struct and override it in the unit test.

Reply:
Alternatively we can keep the function and don't set general default value for testing (it'll be empty for tests and enforce each test to write its own mocking)

Reply:
> Previously it was a const so that it is immutable, and we might want to keep this benefit so that we don't accidentally modify the version number

Makes sense to me! 

> Alternatively we can keep the function and don't set general default value for testing (it'll be empty for tests and we enforce each test to write its own mocking)

I like this too! I kinda like the idea of keeping the mocks within the client testing. (Maybe I'm just not aware of the pattern where you provide a general mock from the source package - is this a thing?)